### PR TITLE
RFCs for flexible config dirs, supports Issue #309

### PR DESCRIPTION
**Pull request by ddobrin**

### CODE REVIEW

**Review by stevehu — APPROVED** [review]
It would be much easier to review and priority individual RFC document instead of three of them. Next time, let's create separate pull requests so that we can speed up the process.

**Comment by ddobrin** [comment]
sure, can be done.

@stevehu : Take also into consideration the fact that issues created to solve each RFC doc should link to the "approved" and merged RFC doc in Git, not the PR per se.

File: light-4j/0012-flexible-config-directory.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    outputFormat: yml
+
+Another use case: configurations, respectively microservice specific data is intended to be loaded from separate directories
+- specifying configurations in multiple folders, in addition to the standard /src/main/resources/config
+
+### Guide-level explanation
+
+```
+The Config.loadMapConfig(String fileName) method can be overloaded with a directory location as:
+  Config.loadMapConfig(String fileName, String directoryName).
```
Comment (bkuberek) [inline light-4j/0012-flexible-config-directory.md:35]: Move the description outside the code block

File: light-4j/0013-abs-rel-config-directory.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+```
+  ...
+  private InputStream getConfigStream(String configFilename) {
+     InputStream inStream = null;
+     try {
+         inStream = new FileInputStream(EXTERNALIZED_PROPERTY_DIR + "/" + configFilename);
+  ...
+```
+
+The implementation should check whether the path starts with a ```/``` and handle the situation of either:
```
Comment (bkuberek) [inline light-4j/0013-abs-rel-config-directory.md]: use single \` in inline code block as opposed to \`\`\`

Comment (bkuberek) [reply light-4j/0013-abs-rel-config-directory.md]: I don't really care so much about this if this a common convention used. I'm just more familiar with single quote for inline code and triple quote for code blocks. Feel free to ignore this.

File: light-4j/0014-support-list-config-directory.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Alternatively, a team might wiosh to separate:
+- configs for all middleware handlers
+- API-specific business related config files
+
+These are just examples.
+
+### Guide-level explanation
+
+The format of "light-4j-config-dir" could be, as an example: ```-Dlight-4j-config-dir='/config:/etc/service:/var/secrets' ```
```
Comment (bkuberek) [inline light-4j/0014-support-list-config-directory.md]: I think you missed a line break after 
> example: ...

**Review by bkuberek — CHANGES_REQUESTED** [review]

**Comment by bkuberek** [comment]
I left some comments regarding the structure of the documents themselves but the content looks good to me.

File: light-4j/0014-support-list-config-directory.md (hunk 3 above)
Comment (bkuberek) [reply light-4j/0014-support-list-config-directory.md]: or maybe not. I noticed now it was inline code. However it wraps on my screen so you might want to consider it being on its own line.

**Review by bkuberek — APPROVED** [review]

**Review by DSchrupert — APPROVED** [review]

File: light-4j/0012-flexible-config-directory.md (hunk 1 above)
Comment (stevehu) [reply light-4j/0012-flexible-config-directory.md:35]: Updated.

File: light-4j/0013-abs-rel-config-directory.md (hunk 2 above)
Comment (stevehu) [reply light-4j/0013-abs-rel-config-directory.md]: updated.

File: light-4j/0014-support-list-config-directory.md (hunk 3 above)
Comment (stevehu) [reply light-4j/0014-support-list-config-directory.md]: updated.

**Pull request merged**